definitions/subworkflows/single_sample_sv_callers.cwl

```diff
@@ -0,0 +1,141 @@
+        type: File?
+        outputSource: run_cnvkit/normal_antitarget_coverage
+    normal_target_coverage:
+        type: File?
+        outputSource: run_cnvkit/normal_target_coverage
+    reference_coverage:
+        type: File?
+        outputSource: run_cnvkit/reference_coverage
+    cn_diagram:
+        type: File?
+        outputSource: run_cnvkit/cn_diagram
+    cn_scatter_plot:
+        type: File?
+        outputSource: run_cnvkit/cn_scatter_plot
+    tumor_antitarget_coverage:
+        type: File
+        outputSource: run_cnvkit/tumor_antitarget_coverage
+    tumor_target_coverage:
+        type: File
+        outputSource: run_cnvkit/tumor_target_coverage
+    tumor_bin_level_ratios:
+        type: File
+        outputSource: run_cnvkit/tumor_bin_level_ratios
+    tumor_segmented_ratios:
+        type: File
+        outputSource: run_cnvkit/tumor_segmented_ratios
+    manta_diploid_variants:
+        type: File?
+        outputSource: run_manta/diploid_variants
+    manta_somatic_variants:
+        type: File?
+        outputSource: run_manta/somatic_variants
+    manta_all_candidates:
+        type: File
+        outputSource: run_manta/all_candidates
+    manta_small_candidates:
+        type: File
+        outputSource: run_manta/small_candidates
+    manta_tumor_only_variants:
+        type: File?
+        outputSource: run_manta/tumor_only_variants
+    smoove_output_variant:
+        type: File
+        outputSource: run_smoove/output_vcf
+
+steps:
+    cram_to_bam:
+        run: ../tools/cram_to_bam.cwl
+        in:
+            cram: cram
+            reference: reference
+        out:
+            [bam]
+    index_bam:
+        run: ../tools/index_bam.cwl
+        in:
+            bam: cram_to_bam/bam
+        out:
+            [indexed_bam]
+    run_cnvkit:
+        run: ../tools/cnvkit_batch.cwl
+        in:
+            access: cnvkit_access
+            bait_intervals: cnvkit_bait_intervals
+            diagram: cnvkit_diagram
+            drop_low_coverage: cnvkit_drop_low_coverage
+            method: cnvkit_method
+            reference_cnn: cnvkit_reference_cnn
+            tumor_bam: index_bam/indexed_bam
+            scatter_plot: cnvkit_scatter_plot
+        out:
+            [intervals_antitarget, intervals_target, normal_antitarget_coverage, normal_target_coverage, reference_coverage, cn_diagram, cn_scatter_plot, tumor_antitarget_coverage, tumor_target_coverage, tumor_bin_level_ratios, tumor_segmented_ratios]
+    run_manta:
+        run: ../tools/manta_somatic.cwl
+        in:
+            call_regions: manta_call_regions
+            non_wgs: manta_non_wgs
+            output_contigs: manta_output_contigs
+            reference: reference
+            tumor_bam: index_bam/indexed_bam
+        out: 
+            [diploid_variants, somatic_variants, all_candidates, small_candidates, tumor_only_variants]
+    run_smoove:
+        run: ../tools/smoove.cwl
+        in:
+            bams: 
```

Comment:
Did you try specifying this more directly without the JS?  It seems like a variation on this ought to work: https://github.com/genome/analysis-workflows/blob/33c06a8096670ed8c291b79e3f3a3b2b8af80533/definitions/subworkflows/pindel_cat.cwl#L40-L41
(Not sure if we'd still need the `MultipleInputFeatureRequirement` even though there's only one source here.)

Reply:
I tried
```
bams: [index_bam/indexed_bam]
```
But when I run `cwltool --validate` on the subworkflow it complains that a source of type `File` is incompatible with sink of type `{"type": "array", "items": "File:}`

Which led me to [this](https://www.biostars.org/p/218153/). And that's how I implemented it. Not sure if there is a better way now. 

I wrote the `smoove.cwl` to handle array of bams to allow the same tool work with single sample and groups. I could change that so that we have a `single_sample_smoove.cwl` and a `cohort_smoove.cwl` and then remove the javascript requirement.

Reply:
Looks like `linkMerge: merge_nested` is what would make the `MultipleInputFeatureRequirement` version work:
https://www.commonwl.org/v1.0/Workflow.html#WorkflowStepInput

I tried it out, so here's apaul7/cancer-genomics-workflow#1 if you like it!

Reply:
I was running the test data through and it failed.
```
No coercion defined from wom value(s) of type 'class 
wom.types.WomMaybePopulatedFileType$' to 
'Array[class wom.types.WomMaybePopulatedFileType$]'
```

Switched to `linkMerged: merge_flattened` and it looks like the inputs are being created correctly for smoove. Testing with the real test data to confirm.

Reply:
test data workflow succeeded with `merge_flattened`